src/mds/Server.cc

```diff
@@ -3839,7 +3839,8 @@ void Server::handle_client_setattr(MDRequestRef& mdr)
 								   truncating_smaller, changed_ranges));
 
   // flush immediately if there are readers/writers waiting
-  if (cur->get_caps_wanted() & (CEPH_CAP_FILE_RD|CEPH_CAP_FILE_WR))
+  if (xlocks.count(&cur->filelock) &&
+      (cur->get_caps_wanted() & (CEPH_CAP_FILE_RD|CEPH_CAP_FILE_WR)))
```

Comment:
On Wed, Jul 27, 2016 at 5:35 AM, Gregory Farnum <notifications@github.com>
wrote:

> In src/mds/Server.cc
> <https://github.com/ceph/ceph/pull/10393#discussion_r72341542>:
>
> > @@ -3839,7 +3839,8 @@ void Server::handle_client_setattr(MDRequestRef& mdr)
> >  								   truncating_smaller, changed_ranges));
> >
> >    // flush immediately if there are readers/writers waiting
> > -  if (cur->get_caps_wanted() & (CEPH_CAP_FILE_RD|CEPH_CAP_FILE_WR))
> > +  if (xlocks.count(&cur->filelock) &&
> > +      (cur->get_caps_wanted() & (CEPH_CAP_FILE_RD|CEPH_CAP_FILE_WR)))
>
> Why do we only want to flush if *we* are xlocking? get_caps_wanted() is
> telling us that *somebody* wants Frw caps they don't have yet, and we
> want to service them, right?
>
No, get_caps_wanted()  return what caps clients want. It's irrelevant to
what caps clients have.

> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ceph/ceph/pull/10393/files/21ee645c8fe62c68caa4612193d494827440dc74..fdf98c10260546fdca13dbe2edc4445faccc13f5#r72341542>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACpZr2KT6Wd90NMmzaneOwsLUyTeBbE9ks5qZn22gaJpZM4JShx2>
> .
>